### PR TITLE
Storage concurrently tests and bug fix

### PR DESCRIPTION
**Pull request by liguozhong**
Storage concurrently tests and bug fix
pr https://github.com/prometheus/prometheus/pull/6770
issue #6714

### CODE REVIEW

**Comment by liguozhong** [comment]
A new fanout directory is created to add tests, because using teststorage.new (T) in the storage directory will cause the following error

# github.com/prometheus/prometheus/storage
FAIL	github.com/prometheus/prometheus/storage [setup failed]
import cycle not allowed in test
package github.com/prometheus/prometheus/storage (test)
	imports github.com/prometheus/prometheus/storage/tsdb
	imports github.com/prometheus/prometheus/storage

**Comment by liguozhong** [comment]
pr #6770
Through the test case, it is found that in the last PR, one line of code is missing, resulting in the result being empty



image

**Comment by liguozhong** [comment]
tests code
image

**Comment by liguozhong** [comment]
tests PASS
image

File: storage/fanout/fanout_test.go [hunk 1]
```diff
@@ -0,0 +1,107 @@
+// Copyright 2017 The Prometheus Authors
```
Comment (brian-brazil) [inline storage/fanout/fanout_test.go]: 2020

File: storage/fanout/fanout_test.go [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+)
+
+func TestSelectSorted(t *testing.T) {
+
+	inputLabel := labels.FromStrings(model.MetricNameLabel, "a")
+	outputLabel := labels.FromStrings(model.MetricNameLabel, "a")
+
+	inputTotalSize := 0
+
+	//0: build test data
```
Comment (brian-brazil) [inline storage/fanout/fanout_test.go]: These comments don't add anything and should be removed.

This style of comment is usually also only when you're doing multiple tests inside one function.

File: storage/fanout/fanout_test.go [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+	priStorage := teststorage.New(t)
+	defer priStorage.Close()
+	// A time series that has two samples and then goes stale.
```
Comment (brian-brazil) [inline storage/fanout/fanout_test.go]: No need for these comments, plus they're incorrect.

File: storage/fanout/fanout_test.go (hunk 1 above)
Comment (liguozhong) [reply storage/fanout/fanout_test.go]: done

File: storage/fanout/fanout_test.go (hunk 3 above)
Comment (liguozhong) [reply storage/fanout/fanout_test.go]: done

File: storage/fanout/fanout_test.go (hunk 2 above)
Comment (liguozhong) [reply storage/fanout/fanout_test.go]: done

**Comment by brian-brazil** [comment]
Thanks!

**Pull request merged**